src/main/resources/templates/fragments/navBarSubMenu.html

```diff
@@ -6,18 +6,21 @@
 </head>
 <body>
 
+<!-- This is expected to be called in the context of a list -->
 <div th:fragment="navBarSubMenu (menu)" th:remove="tag">
-  <!-- This is expected to be called in the context of a list -->
-  <li class="app-subnav__section-item">
+  <th:block
```

Comment:
This brings the business logic into the view. That's really a dangerous design practice.

It would be better if the model understands the condition and then returns only viewable items.

On any given page display, the menu won't change.